### PR TITLE
alternative way to sort addresses in a registrar using timestamp

### PR DESCRIPTION
**Pull request by voith**
### What was wrong?

`find_deploy_block_number` doesn't work as expected in all cases, especially when using with `py-evm`. 

### How was it fixed?
Since `find_deploy_block_number` is only used to sort contracts based on deployment order, use a timestamp approach instead and remove `find_deploy_block_number `.
[MemoryBackend](https://github.com/ethereum/populus/blob/d18ac1785cc575671baada62e594d7ab7997003d/populus/contracts/backends/memory.py#L8) and [JSONFileBackend](https://github.com/ethereum/populus/blob/d18ac1785cc575671baada62e594d7ab7997003d/populus/contracts/backends/filesystem.py#L41) are the only two registrars with `is_registrar = True`.
I have modified both to return a tuple of (address, timestamp). This is later sorted in the `Registar` class.

However, I didn't spend much time on `JSONFileBackend` and overall this PR is very dirty!
I'm open to any sort of feedback!

### CODE REVIEW

File: populus/utils/epoch.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import time
+
+
+def epoc_time_now():
+    return time.time()
```
Comment (pipermerriam) [inline populus/utils/epoch.py]: We might want to consider truncating this to the integer representation since `time.time()` returns a float.

**Comment by voith** [comment]
@pipermerriam addressed all your comments. Let me know if anything else is needed. Else I'd like to get this merged.
> In case it's not clear, this type of registry will be more appropriately handled using ethpm packages 

Thanks, good to know!

**Comment by voith** [comment]
@pipermerriam Can we merge this if you think this is good enough?

**Comment by pipermerriam** [comment]
@voith :shipit:

**Comment by voith** [comment]
Thanks @pipermerriam

**Comment by voith** [comment]
@pipermerriam Thank you so much for taking the time to review this. If anything goes wrong because of this PR, I'll fix it myself.

**Pull request merged**